Create ui-c8y-1020-26-5-removing-not-used-error-actual-role-can-be-remove-even-is.md (#2448)
SoftwareAG/c8y-docs: content/change-logs/platform-services/ui-c8y-1020-26-5-removing-not-used-error-actual-role-can-be-remove-even-is.md

---
date: ""

> title: Correct error message when deleting an inventory role and a global role.

Reviewer: ```suggestion
title: Correct error message when deleting an inventory role or a global role
```

> title: The display of errors returned by the server during the removal of a global role has been corrected.

Reviewer: ```suggestion
title: Error message during the removal of a global role provides correct information
```

Reviewer: @kgol-c8y Shouldn´t this refer to "inventory role" instead of global role?

Reviewer 2: ```suggestion
title: Error message on removal of an inventory access mapping in SSO configuration provides correct information
```

> title: Error message on removal of an inventory access mapping in SSO configuration provides correct information

Reviewer: @kgol-c8y Are you sure that this is what the ticket is about? I think this is just an example which caused the fix to happen but not the fix itself. Below, the text also says global role, is this correct?

Reviewer 2: Thats true @BeateRixen I looked into the ticket as well. As per my understanding I think, the changes is for both global roles and inventory roles. I am trying to confirm the same with Marcin quickly and update here.

Reviewer 2: ```suggestion
title: Correct error message when deleting an inventory role and a global role.
```

> title: removing not used error - actual role can be remove even is assigned to user (#6856) [GRAFT][release/cd] (#7054)

Author: The Claude AI suggested to use the following title:
```suggestion
title: Allow removing roles that are assigned to users
```

product_area: Platform services
change_type:
  - value: change-VSkj2iV9m
    label: Fix
component:
  - value: q3kclF6pO
    label: Authentication
build_artifact:
  - value: tc-pjJiURv9Y
    label: ui-c8y
ticket: MTM-53701
version: 1020.26.5
---

> removing not used error - actual role can be remove even is assigned to user (#6856) [GRAFT][release/cd] (#7054)

Author: The Claude AI suggested to use the following body:
```suggestion
Previously, it was not possible to remove a role if it was still assigned to a user, even if the role was no longer used. With this change, administrators can now remove roles that are assigned to users. This allows cleaning up unused roles and simplifies role management. Existing roles assigned to users can now be safely removed without impacting the users.
```

> In the event of an error while deleting a global role or an inventory role, the message always stated that the role couldn't be deleted because it was assigned to a user. This fix ensures that the error message will now reflect the actual cause of the problem.

Reviewer: ```suggestion
In the event of an error while deleting an inventory role or a global role, the message always stated that the role couldn't be deleted because it was assigned to a user. This fix ensures that the error message will now reflect the actual cause of the problem.
```

> If the server returned an "unprocessable content" error while removing a global role, the application always indicated that the issue was due to the role being assigned to a user. Now, the application will display the correct error.

Reviewer 2: the release notes content here is different from ticket release notes 
![image](https://github.com/user-attachments/assets/44bf919e-0f1b-471e-b8f9-6cb2d4fe4731)
can you update same information at both places?

Reviewer: This is a different process. The automation takes the text from the commit message. In the tickets, the text is added manually. Actually, the fields are no longer used but were only relevant for releases 10.18 and beyond.

> If the server returned an "unprocessable content" error while removing a global role, the application always indicated that the issue was due to the role being assigned to a user. Now, the application will display the correct error.

Reviewer: ```suggestion
If the server returned an "unprocessable content" error while removing a global role, the error message always indicated that the issue occurred because the global role was assigned to a user. Now, the error message will display the correct information, for example "Role is used in inventory access mapping in SSO configuration".
```

> If the server returned an "unprocessable content" error while removing a global role, the error message always indicated that the issue occurred because the global role was assigned to a user. Now, the error message will display the correct information, for example "Role is used in inventory access mapping in SSO configuration".

Reviewer 2: ```suggestion
In the event of an error while deleting a global role or an inventory role, the message always stated that the role couldn't be deleted because it was assigned to a user. This fix ensures that the error message will now reflect the actual cause of the problem.
```
